### PR TITLE
[SPLTRP-0915]: tech-debt: Address SonarQube code smell and improve code coverage for classes below 80%

### LINKED ISSUE #915: tech-debt: Address SonarQube code smell and improve code coverage for classes below 80%
## Summary

Address SonarQube findings: 1 code smell and several classes with code coverage below the 80% minimum target.

---

## 1. Code Smell: `CurrencyExtensions.kt` — Too many `when` branches

**File:** `core/design-system/src/main/kotlin/es/pedrazamiguez/splittrip/core/designsystem/extension/CurrencyExtensions.kt`

**Finding:** Reduce the number of `when` branches from 31 to at most 30.

**Recommendation:** This is likely a reasonable pattern for exhaustive currency-to-resource mapping. Options:
1. **Suppress / exclude** — Add a Sonar exclusion or `@SuppressWarnings` if the `when` expression is an exhaustive mapping that is inherently 1:1 with the enum values. This is the recommended approach since the number of branches will grow as currencies are added, and refactoring to a `Map` would reduce readability for no real benefit.
2. **Refactor to `Map` lookup** — Replace the `when` block with a `Map<Currency, Int>` constant. This satisfies the rule but may be less idiomatic for resource ID mapping.

> **Note:** This overlaps with Detekt alert [#388](https://github.com/pedrazamiguez/split-trip/security/code-scanning/388) tracked in #914. If the `when` block is addressed there, this Sonar finding may also be resolved.

---

## 2. Code Coverage Below 80% Minimum

The project is above 80% overall, but the following individual classes are below the target:

### Critical (0% coverage — no tests at all)

| File | Coverage | Lines | Branches |
|---|---|---|---|
| `domain/.../usecase/setting/GetSelectedGroupCurrencyUseCase.kt` | 0.0% | 2 missed | 0 |
| `domain/.../usecase/setting/SetSelectedGroupUseCase.kt` | 0.0% | 2 missed | 0 |

These are small use cases that likely delegate to `PreferenceRepository`. They should be straightforward to test.

### Below 80% threshold

| File | Coverage | Lines Missed | Branches Missed |
|---|---|---|---|
| `features/withdrawals/.../handler/WithdrawalCurrencyHandler.kt` | 58.8% | 2 | 5 |
| `data/.../sync/CloudSyncDelegates.kt` | 78.5% | 10 | 4 |

### Above 80% but with gaps worth addressing

| File | Coverage | Lines Missed | Branches Missed |
|---|---|---|---|
| `features/groups/.../mapper/impl/GroupUiMapperImpl.kt` | 86.4% | 0 | 3 |
| `features/main-entry/.../viewmodel/MainViewModel.kt` | 88.9% | 2 | 0 |
| `features/contributions/.../handler/ContributionConfigHandler.kt` | 92.6% | 2 | 6 |
| `features/contributions/.../handler/ContributionSubmitHandler.kt` | 93.8% | 2 | 2 |
| `data/.../repository/impl/ContributionRepositoryImpl.kt` | 94.7% | 0 | 1 |
| `data/.../repository/impl/ExpenseRepositoryImpl.kt` | 94.7% | 0 | 1 |

---

### Priority 1 — Bring 0% classes to ≥80%
1. **`GetSelectedGroupCurrencyUseCase`** — Add unit test covering the delegation to `PreferenceRepository`.
2. **`SetSelectedGroupUseCase`** — Add unit test covering the delegation to `PreferenceRepository`.

### Priority 2 — Bring <80% classes to ≥80%
3. **`WithdrawalCurrencyHandler`** (58.8%) — Add tests for uncovered branches (likely edge cases in currency conversion, error handling paths). Note: some coverage may have improved with #896 fix adding comma-decimal tests.
4. **`CloudSyncDelegates.kt`** (78.5%) — Add tests for uncovered sync edge cases (error paths, race conditions).

### Priority 3 — Close remaining gaps (nice-to-have)
5. Address the remaining classes above 80% if there are easy-to-cover branches.

---

## Acceptance Criteria

- [ ] `GetSelectedGroupCurrencyUseCase` coverage ≥ 80%
- [ ] `SetSelectedGroupUseCase` coverage ≥ 80%
- [ ] `WithdrawalCurrencyHandler` coverage ≥ 80%
- [ ] `CloudSyncDelegates.kt` coverage ≥ 80%
- [ ] `CurrencyExtensions.kt` Sonar code smell addressed (suppressed with justification or refactored)
- [ ] No reduction in overall project coverage
- [ ] All new tests use JUnit 5 + MockK (no Kotlin `assert()`)

### PR DESCRIPTION
**Pull request by pedrazamiguez**
## 🔗 Related Issues

Closes #915 

## 🧪 Testing Instructions

- [ ] I have tested these changes locally
- [ ] I have added/updated tests as needed
- [ ] All existing tests pass

## ✅ Quality Checklist

- [ ] My code follows the project's coding standards
- [ ] I have performed a self-review of my code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] My changes generate no new warnings
- [ ] I have updated documentation as needed

### CODE REVIEW

File: data/src/test/kotlin/es/pedrazamiguez/splittrip/data/sync/CloudSyncDelegatesTest.kt [hunk 1]
```diff
@@ -304,5 +374,24 @@ class CloudSyncDelegatesTest {
...
+                    scope = this,
+                    entityId = "entity-1",
+                    cloudDelete = { throw CancellationException("Delete cancelled") },
+                    entityLabel = "test"
+                )
+                advanceUntilIdle()
+
+                // CancellationException is rethrown inside the launched coroutine,
+                // which is then cancelled. No further state mutations occur.
+                // The test passes if no unhandled exception propagates to the test scope.
```
[the remark this answers is not in the document]
Comment (pedrazamiguez) [reply data/src/test/kotlin/es/pedrazamiguez/splittrip/data/sync/CloudSyncDelegatesTest.kt]: You're right — this is a valid concern. The test name claims "rethrows CancellationException" but the body has no assertion that distinguishes rethrow from swallow.

This is structurally different from the `syncCreateToCloud` CancellationException test, which IS meaningful: it asserts `updateSyncStatus` is never called with any `SyncStatus`, which would fail if `CancellationException` were swallowed by `catch (e: Exception)` (the SYNC_FAILED path would execute). There's an injectable callback to mock.

`syncDeletionToCloud` has no such callback in the error path — only `Timber.w`, which isn't mockable in unit tests. That left the test with no distinguishing assertion.

Fixing this by capturing the `Job` launched inside `syncDeletionToCloud` and asserting `job.isCancelled == true` after `advanceUntilIdle()`. With `StandardTestDispatcher` the coroutine is queued (not yet started) immediately after the call, so the child Job is registered and capturable. After advancing:
- If CancellationException is rethrown → coroutine is cancelled → `isCancelled = true` ✓
- If CancellationException is swallowed by `catch (e: Exception)` → coroutine completes normally → `isCancelled = false` ✗ (test would correctly fail)

Renaming + rewriting the test accordingly.
